include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -589,5 +589,9 @@ ERROR(experimental_not_supported_in_production,none,
       "experimental feature '%0' cannot be enabled in production compiler",
       (StringRef))
 
+GROUPED_WARNING(Ounchecked_with_strict_safety,Unsafe,none,
+      "'-Ounchecked' is not memory-safe and should not be combined with "
```

Comment:
I wonder what do you think about `-disable-access-control`. It might be used to circumvent some safety checks, but it is a frontend only flag. Should we also diagnose something like that?